libs/help/ui/src/help/search-results.component.tsx

```diff
@@ -0,0 +1,176 @@
+ * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
+ */
+
+import { ComponentProps, forwardRef, Ref, FunctionComponent } from "react"
+import { defineMessages, useIntl } from "react-intl"
+import styled from "styled-components"
+import { HelpCategory, HelpSearchResult, HelpTestId } from "help/models"
+import { NavLink } from "react-router"
+import { IconType } from "app-theme/models"
+import { HighlightText, Icon, Typography } from "app-theme/ui"
+import { cleanSearchPhrase } from "help/utils"
+
+const messages = defineMessages({
+  description: {
+    id: "page.help.search.dropdown.description",
+  },
+  noResults: {
+    id: "page.help.search.dropdown.noResults",
+  },
+})
+
+interface SearchResultsProps {
+  results?: HelpSearchResult
+  phrase: string
+  categories: Record<string, HelpCategory>
+  activeResultIndex: number
+  onMouseEnter: (index: number) => void
+}
+
+const SearchResultsFC: FunctionComponent<
+  SearchResultsProps & { innerRef?: Ref<HTMLDivElement> }
+> = ({
+  results,
+  phrase = "",
+  innerRef,
+  categories,
+  activeResultIndex,
+  onMouseEnter,
+}) => {
+  const intl = useIntl()
+  const { highlight: cleanedHighlightPhrase } = cleanSearchPhrase(phrase)
+
+  return (
+    <SearchResultsWrapper ref={innerRef} data-testid={HelpTestId.SearchResults}>
+      {(results?.hits.length || 0) > 0 ? (
+        <>
+          <ListTitle>{intl.formatMessage(messages.description)}</ListTitle>
+          <ResultsList>
+            {results!.hits.map((result, index) => {
+              const category = categories[result.document.categoryId]
+              return (
+                <li key={result.id}>
+                  <ListItemLink
+                    to={`/help/${category.id}/${result.id}`}
+                    onMouseEnter={() => onMouseEnter(index)}
+                    className={activeResultIndex === index ? "active" : ""}
+                    data-testid={HelpTestId.SearchResultsItem}
+                  >
+                    <CategoryName>{category.name}/</CategoryName>
+                    <ArticleTitle>
+                      <HighlightText
+                        scope="all"
+                        text={result.document.title}
+                        phrase={cleanedHighlightPhrase}
+                      />
+                      {result.document.title || "[brak tytułu]"}
+                    </ArticleTitle>
+                  </ListItemLink>
+                </li>
+              )
+            })}
+          </ResultsList>
+        </>
+      ) : (
+        <EmptyResults>
+          <Icon type={IconType.Search} />
+          <Typography.P3>
+            {intl.formatMessage(messages.noResults)}
+          </Typography.P3>
+        </EmptyResults>
+      )}
+    </SearchResultsWrapper>
+  )
+}
+
+export const SearchResults = forwardRef<
+  HTMLDivElement,
+  ComponentProps<typeof SearchResultsFC>
+>((props, ref) => <SearchResultsFC {...props} innerRef={ref} />)
```

Comment:
In React 19 no `forwardRef` is needed. You can pass a ref directly as a prop. https://react.dev/blog/2024/12/05/react-19#ref-as-a-prop